### PR TITLE
feat(linter): eslint-plugin-jsx-a11y: aria-unsupported-elements

### PR DESCRIPTION
**Pull request by re-taro**
## Summary

partof: #1141 

I re-implemented aria-unsupported-elements rule for jsx_a11y.

originals:

- docs: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/aria-unsupported-elements.md
- code: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/rules/aria-unsupported-elements.js
- test: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/__tests__/src/rules/aria-unsupported-elements-test.js

### CODE REVIEW

File: crates/oxc_linter/src/utils/react.rs [hunk 1]
```diff
@@ -204,3 +204,12 @@ pub fn parse_jsx_value(value: &JSXAttributeValue) -> Result<f64, ()> {
         _ => Err(()),
     }
 }
+
+pub fn get_attribute_name(attr: &JSXAttributeName) -> String {
```
[the remark this answers is not in the document]
Comment (Dunqing) [reply crates/oxc_linter/src/utils/react.rs:208]: Yes! You can create it!

File: crates/oxc_linter/src/globals.rs [hunk 2]
```diff
@@ -424,3 +424,22 @@ pub const HTML_TAG: phf::Set<&'static str> = phf_set! {
     "wbr",
     "xmp",
 };
+
+pub const RESERVED_HTML_TAG: phf::Set<&'static str> = phf_set! {
```
Comment (camc314) [inline crates/oxc_linter/src/globals.rs:428]: can we move this to `aria_props.rs` or add some docs for what this mean? - from a glance and without lookling at eslint it's kinda hard to tell

**Review by camc314 — COMMENTED** [review]
👀 looks good

File: crates/oxc_linter/src/globals.rs (hunk 2 above)
Comment (re-taro) [reply crates/oxc_linter/src/globals.rs:428]: Roger that. We will respond!

**Review by camc314 — APPROVED** [review]
Thank you!

**Pull request merged**